base/linalg/diagonal.jl

```diff
@@ -234,6 +234,11 @@ A_mul_B!(out::AbstractMatrix, A::Diagonal, in::AbstractMatrix) = out .= A.diag .
 Ac_mul_B!(out::AbstractMatrix, A::Diagonal, in::AbstractMatrix) = out .= ctranspose.(A.diag) .* in
 At_mul_B!(out::AbstractMatrix, A::Diagonal, in::AbstractMatrix) = out .= transpose.(A.diag) .* in
 
+# ambiguities with Symmetric/Hermitian
+A_mul_Bt(A::Diagonal, B::RealHermSymComplexSym) = A*B
+At_mul_B(A::RealHermSymComplexSym, D::Diagonal) = A*B
+A_mul_Bc(A::Diagonal, B::RealHermSymComplexHerm) = A*B
+Ac_mul_B(A::RealHermSymComplexHerm, D::Diagonal) = A*B
```

Comment:
> Are these definitions correct (considering the recursive nature of `[c]transpose`)?

I didn't think about that... The `t` in `At_mul_B` methods are also considered to be recursive, right?

> and given the second input was named D, not B - this method must not have been covered in tests?

Should never do changes like that in the last minute... Will add some tests.